extensions/kubeapi/projects/update.go

```diff
@@ -0,0 +1,45 @@
+package projects
+
+import (
+	"context"
+
+	v3 "github.com/rancher/rancher/pkg/apis/management.cattle.io/v3"
+	"github.com/rancher/shepherd/clients/rancher"
+	"github.com/rancher/shepherd/extensions/unstructured"
+	"github.com/rancher/shepherd/pkg/api/scheme"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// UpdateProject is a helper function that uses the dynamic client to update a project in a specific cluster.
+func UpdateProject(client *rancher.Client, updatedProject *v3.Project) (*v3.Project, error) {
+	dynamicClient, err := client.GetDownStreamClusterClient(LocalCluster)
+	if err != nil {
+		return nil, err
+	}
+
+	projectResource := dynamicClient.Resource(ProjectGroupVersionResource).Namespace(updatedProject.Namespace)
```

Comment:
this might error out if a user tries to update the name as they will get no project found. probably we might want to change this part. this applies to other update functions as well.

Reply:
This function is to update an existing Project. If a project name is not found, it will error out.